xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/pdf/impl/PdfExportImpl.java

```diff
@@ -329,26 +329,14 @@ String applyCSS(String html, String css, XWikiContext context)
             XHTMLDocumentFactory docFactory = XHTMLDocumentFactory.getInstance();
             SAXReader reader = new SAXReader(docFactory);
 
-            // Dom4J 2.1.1 disables external DTDs by default, which is required here, putting it back.
+            // Dom4J 2.1.1 disables external DTDs by default, so we set our own XMLReader.
             // See https://github.com/dom4j/dom4j/issues/51
+            XMLReader xmlReader = Utils.getComponent(XMLReaderFactory.class).createXMLReader();
             try {
-                reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true);
-            } catch (SAXNotSupportedException | SAXNotRecognizedException e) {
-                // We ingore these 2 exceptions to properly handle the case when a non-Xerces parser is provided by
-                // JAXP. Here are the possible flows:
-                //
-                // A) JAXP gives a Xerces instance:
-                // dom4j 2.1.1 sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to false,
-                // to disable DTD. No exception is thrown.
-                // XWiki sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to true, to enable DTD
-                // loading. No exception.
-                //
-                // B) JAXP gives a non-Xerces instance:
-                // dom4j 2.1.1 sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to false. The
-                // feature is not recognized: exception is thrown and ignored.
-                // XWiki sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to true. Exception is
-                // thrown (and was not being ignored). Boom.
+                xmlReader.setFeature("http://xml.org/sax/features/namespaces", true);
+            } catch (SAXNotRecognizedException | SAXNotSupportedException e) {
```

Comment:
And also explain why it's ok to ignore these exceptions and what will be the consequences?

Reply:
IIRC all current parsers support this (and the default value for that feature is `true` anyway), so all this is basically a no-op in practice, which is kept for backwards compatibility.

Reply:
cool so let's add this as comment!

BTW checkstyle is supposed to fail on empty blocks (not sure when we lost that check, I'll need to do some archeology ;)).

Reply:
> IIRC all current parsers support this (and the default value for that feature is true anyway)

So why do we need it?